src/pre-engagement-form/index.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/* eslint-disable react/require-default-props */
+import React from 'react';
+import { connect } from 'react-redux';
+import { useForm, FormProvider } from 'react-hook-form';
+import * as FlexWebChat from '@twilio/flex-webchat-ui';
+
+import { AseloWebchatState } from '../aselo-webchat-state';
+import type { PreEngagementForm as PreEngagementFormDefinition } from './form-components/types';
+import { generateForm } from './form-components';
+import { LocalizationProvider } from './localization';
+import SubmitButton from './form-components/submit-button';
+import Title from './form-components/title';
+import { resetForm } from './state';
+
+export { PreEngagementFormDefinition };
+
+export const EMAIL_PATTERN = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
+
+type Props = {
+  manager: FlexWebChat.Manager;
+} & ReturnType<typeof mapStateToProps> &
+  typeof mapDispatchToProps;
+
+const PreEngagementForm: React.FC<Props> = ({ formState: defaultValues, formDefinition, manager, resetFormAction }) => {
+  const methods = useForm({ defaultValues, mode: 'onChange' });
+  const { handleSubmit, formState } = methods;
+  const { isValid } = formState;
+
+  const onSubmit = handleSubmit((data) => {
+    const payload = { formData: data };
+    console.log({ payload });
+    FlexWebChat.Actions.invokeAction('StartEngagement', payload);
+    resetFormAction();
+  });
+
+  if (formDefinition === undefined) {
+    return null;
+  }
+
+  return (
+    <FormProvider {...methods}>
+      <LocalizationProvider manager={manager}>
+        <form className="Twilio-DynamicForm" onSubmit={onSubmit}>
+          <Title title={formDefinition.description} />
+          {generateForm(formDefinition.fields)}
+          {formDefinition.submitLabel && <SubmitButton label={formDefinition.submitLabel} disabled={!isValid} />}
```

Comment:
I think if `formDefinition.submitLabel` is missing we should either prevent the app from starting or default to something (like the English button).

Reply:
This is intentional (for now). Currently, closed hours and holidays are "forms" as well, but without a submit button.

Now that we will have full control of pre-engagement, we can think of better ways to display them. But for now, I've kept them being defined just like they were (but without the ugly workaround of a required field that's invisible 😂)